Cleanup NEST Server environment variable handling (#2942)
nest/nest-simulator: doc/htmldoc/connect_nest/nest_server.rst

from the NEST Docker image. Please check out the corresponding
:ref:`installation instructions <docker>` for more details.

.. _sec_server_vars:

Set environment variables for security options
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

NEST Server comes with a number of access restrictions that are meant to protect your
computer. After careful consideration, each of the restrictions can be disabled by setting
a corresponding environment variable.

> * ``NEST_SERVER_DISABLE_AUTH``: By default, the NEST Server requires a NESTServerAuth tokens. Setting this variable to ``0`` disables this restriction. A token is automatically created and printed to the console by NEST Server upon start-up. If needed, a custom token can be set using the environment variable  ``NEST_SERVER_ACCESS_TOKEN``

Author: ```suggestion
* ``NEST_SERVER_DISABLE_AUTH``: By default, the NEST Server requires a NESTServerAuth tokens. Setting this variable to ``1`` disables this restriction. A token is automatically created and printed to the console by NEST Server upon start-up. If needed, a custom token can be set using the environment variable  ``NEST_SERVER_ACCESS_TOKEN``
```

* ``NEST_SERVER_CORS_ORIGINS``: By default, the NEST Server only allows requests from localhost (see `CORS <https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS>`_). Other hosts can be explicitly allowed by supplying them in the form `http://host_or_ip`` to this variable.

> * ``NEST_SERVER_ENABLE_EXEC_CALL``: By default, NEST Server only allows calls to its PyNEST-like API. If the use-case requires the execution of scripts via the ``/exec`` route, this variable can be set to 0. PLEASE BE AWARE THAT THIS OPENS YOUR COMPUTER TO REMOTE CODE EXECUTION.

Author: ```suggestion
* ``NEST_SERVER_ENABLE_EXEC_CALL``: By default, NEST Server only allows calls to its PyNEST-like API. If the use-case requires the execution of scripts via the ``/exec`` route, this variable can be set to ``1``. PLEASE BE AWARE THAT THIS OPENS YOUR COMPUTER TO REMOTE CODE EXECUTION.
```

> * ``NEST_SERVER_DISABLE_RESTRICTION``: By default, NEST Server runs all code passed to the ``/exec`` route through RestrictedPython to sanitize it. To disable this mechanism, this variable can be set to 0. For increased security, code passed in this way only allows explictly whitelisted modules to be imported. To import modules, the variable ``NEST_SERVER_MODULES`` can be set to a standard Python import line like this:

Author: ```suggestion
* ``NEST_SERVER_DISABLE_RESTRICTION``: By default, NEST Server runs all code passed to the ``/exec`` route through RestrictedPython to sanitize it. To disable this mechanism, this variable can be set to ``1``. For increased security, code passed in this way only allows explictly whitelisted modules to be imported. To import modules, the variable ``NEST_SERVER_MODULES`` can be set to a standard Python import line like this:
```

from the NEST Docker image. Please check out the corresponding
:ref:`installation instructions <docker>` for more details.

.. _sec_server_vars:

> Set environment variables for security options
> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 
> To use NEST Server, there are several environment variables users need to set in their environment.
> 
> * Requests require NESTServerAuth tokens. By default, the authentication is on (``NEST_SERVER_DISABLE_AUTH=0``).
> * NEST Server generates a token automatically, but it can also take custom tokens (``NEST_SERVER_ACCESS_TOKEN='alongaccesstoken'``).
> * The `CORS <https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS>`_ origins are restricted. By default, the only allowed CORS origin is ``http://localhost``
>   (``NEST_SERVER_CORS_ORIGINS=http://localhost``).
> * Only API calls are enabled. By default, the exec call is disabled (``NEST_SERVER_ENABLE_EXEC_CALL=0``).
> * The code execution is restricted. By default, the restriction is activated (``NEST_SERVER_DISABLE_RESTRICTION=0``).
> * For security reasons the exec call in NEST Server accepts only modules from NEST_SERVER_MODULES.
> 
>   For example:
> 
>        ``NEST_SERVER_MODULES='import nest; import numpy as np; from numpy import random'``
> 

Reviewer: My rationale here is that it should be not easy for users to copy-paste working settings from this text, but force them to read the text about the implications and piece the setting together manually.

Also, I think the original text was not very clear about what the default setting is and what the change away from that actually implies.

```suggestion
Set environment variables for security options
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

NEST Server comes with a number of access restrictions that are meant to protect your
computer. After careful consideration, each of the restrictions can be disabled by setting
a corresponding environment variable.

* ``NEST_SERVER_DISABLE_AUTH``: By default, the NEST Server requires a NESTServerAuth tokens. Setting this variable to ``0`` disables this restriction. A token is automatically created and printed to the console by NEST Server upon start-up. If needed, a custom token can be set using the environment variable  ``NEST_SERVER_ACCESS_TOKEN``
* ``NEST_SERVER_CORS_ORIGINS``: By default, the NEST Server only allows requests from localhost (see `CORS <https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS>`_). Other hosts can be explicitly allowed by supplying them in the form `http://host_or_ip`` to this variable.
* ``NEST_SERVER_ENABLE_EXEC_CALL``: By default, NEST Server only allows calls to its PyNEST-like API. If the use-case requires the execution of scripts via the ``/exec`` route, this variable can be set to 0. PLEASE BE AWARE THAT THIS OPENS YOUR COMPUTER TO REMOTE CODE EXECUTION.
* ``NEST_SERVER_DISABLE_RESTRICTION``: By default, NEST Server runs all code passed to the ``/exec`` route through RestrictedPython to sanitize it. To disable this mechanism, this variable can be set to 0. For increased security, code passed in this way only allows explictly whitelisted modules to be imported. To import modules, the variable ``NEST_SERVER_MODULES`` can be set to a standard Python import line like this:
       ``NEST_SERVER_MODULES='import nest; import scipy as sp; from numpy import random'``
```

Author: @jougs Thank you for the review. There env variables are set to 0, by default. I will correct this text.
